pkg/inject/inject.go

```diff
@@ -209,49 +209,45 @@ func (conf *ResourceConfig) ParseMetaAndYAML(bytes []byte) (*Report, error) {
 // If injectProxy is false, only the config.linkerd.io annotations are set.
 func (conf *ResourceConfig) GetPatch(injectProxy bool) ([]byte, error) {
 
-	if conf.requireIdentityOnInboundPorts() != "" && conf.identityContext() == nil {
-		return nil, fmt.Errorf("%s cannot be set when identity is disabled", k8s.ProxyRequireIdentityOnInboundPortsAnnotation)
+	// Make a copy of Values and mutate that
+	var copyValues linkerd2.Values
+	copyRawValues, err := yaml.Marshal(conf.values)
+	if err != nil {
+		return nil, err
 	}
 
-	if conf.destinationGetNetworks() != "" {
-		for _, network := range strings.Split(strings.Trim(conf.destinationGetNetworks(), ","), ",") {
-			if _, _, err := net.ParseCIDR(network); err != nil {
```

Comment:
Same here, the pod is failing at runtime instead of stopping injection if the annotation value is invalid.